src/libcore/ptr.rs

```diff
@@ -946,6 +946,56 @@ pub unsafe fn write_volatile<T>(dst: *mut T, src: T) {
     intrinsics::volatile_store(dst, src);
 }
 
+/// Freezes `count * size_of::<T>()` bytes of memory, converting undefined data into
+/// defined-but-arbitrary data.
+///
+/// Uninitialized memory has undefined contents, and interation with that data
+/// can easily cause undefined behavior. This function "freezes" memory
+/// contents, converting uninitialized memory to initialized memory with
+/// arbitrary conents so that use of it is well defined.
```

Comment:
```suggestion
/// arbitrary contents so that use of it is well defined.
```

Reply:
"arbitrary but fixed contents" might be a better formulation.

Also it might be worth noting that use is only well defined for integer type -- even with arbitrary but fixed contents, using this with `bool` or `&T` is UB.

Reply:
> Also it might be worth noting that use is only well defined for integer type -- even with arbitrary but fixed contents, using this with `bool` or `&T` is UB.

That's a *great* point! Definitely worth noting.

Reply:
I have this phrased as "Every bit representation of `T` must be a valid value", but I don't think that's the best way of saying that. Ideas?

Reply:
> I have this phrased as "Every bit representation of `T` must be a valid value",

Is there perhaps an auto-trait waiting to be invented for that? We could ostensibly make the API a bit safer by adding a constraint `T: TheTrait`...?

Reply:
There's been talk of this kind of thing for quite a while (`pub auto trait Pod {}`), but I think that'd be relevant as part of a safe interface over this specific function.

Reply:
I don't think this can be an auto trait -- auto traits are always implemented for fieldless enums, but this function is not sufficient to make a fieldless enum valid.